### DISCUSSION TITLE
Use plural for root directories

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#506] [Ideas]
I have always found it a bit odd that Foundry went with singular-form nouns for `script` and `test`. In software engineering, plurals are more common (and, I would argue, more natural since multiple files are confined in these directories).

I've recently seen that Maple used the plural form:

- https://github.com/maple-labs/maple-core-v2/tree/00f01ae7175885f8d49ac201a1c72465e320b2f6/scripts
- https://github.com/maple-labs/maple-core-v2/tree/00f01ae7175885f8d49ac201a1c72465e320b2f6/tests

What if did the same, and renamed `script` to `scripts`, and `test` to `tests`?

**Comment by smol-ninja** [comment]
I like the idea. But what about the `src` folder? `srcs` (pronounced as `sources`) wouldn't sound great. 

Maple didn't follow the foundry nomenclature as they are using `contracts` folder. So plural goes well with them. imo if we want to use plural names then we should also rename `src` to `contracts`.

**Reply by PaulRBerg** [reply]
Glad you like it.

I think that `src` is such a ubiquitous term in computer science that it's fine to leave it as is.

`contracts` would be a bit confusing because the precompiles and the tests are contracts, too.

**Comment by PaulRBerg** [comment]
Should we proceed with renaming `test` to `tests`, etc.?

@sablier-labs/solidity

**Reply by smol-ninja** [reply]
I've nothing against it. So we can proceed. But lets give it a priority 3, since its only related to tests.

**Reply by andreivladbrg** [reply]
same, don't have any particular opinion

agree with the low priority

**Comment by andreivladbrg** [comment]
https://github.com/sablier-labs/v2-core/issues/1036
